### PR TITLE
custom search view, change icon, add total video and users count

### PR DESCRIPTION
**Pull request by bogdanStepanyuk**

### CODE REVIEW

File: app/src/main/java/com/uptech/example/vidme/adapters/UsersSearchListAdapter.java [hunk 1]
```diff
@@ -52,6 +52,8 @@ public void onBindViewHolder(ViewHolder holder, int position) {
         }
         holder.views.setText(String.valueOf(user.getVideosViews()));
         holder.followers.setText(String.valueOf(user.getFollowerCount()));
+
```
Comment (Dima564) [inline app/src/main/java/com/uptech/example/vidme/adapters/UsersSearchListAdapter.java]: Remove extra space :space_invader:

File: app/src/main/java/com/uptech/example/vidme/adapters/VideoSearchListAdapter.java [hunk 2]
```diff
@@ -53,6 +53,16 @@ public void onBindViewHolder(ViewHolder holder, int position) {
         }
         holder.likesCount.setText(String.valueOf(video.getLikesCount()));
         holder.viewsCount.setText(String.valueOf(video.getViewCount()));
+
+        holder.likesCount.setOnClickListener(new View.OnClickListener() {
```
Comment (Dima564) [inline app/src/main/java/com/uptech/example/vidme/adapters/VideoSearchListAdapter.java]: Use retrolambda

File: app/src/main/java/com/uptech/example/vidme/ui/search/SearchActivity.java [hunk 3]
```diff
@@ -50,52 +60,87 @@ protected void onCreate(Bundle savedInstanceState) {
...
+                editSearch.setHint(R.string.search_hint);
+                editSearch.setCursorVisible(true);
+                searchOff.setVisible(true);
+                searchOn.setVisible(false);
+                editSearch.setVisibility(View.VISIBLE);
+                editSearch.requestFocus();
+                openKeyboard();
+                return true;
+            case R.id.search_button_off:
+                if (!TextUtils.isEmpty(editSearch.getText().toString())) {
```
Comment (Dima564) [inline app/src/main/java/com/uptech/example/vidme/ui/search/SearchActivity.java]: toString() is unnecessary

File: app/src/main/java/com/uptech/example/vidme/ui/search/SearchActivity.java [hunk 4]
```diff
@@ -141,12 +191,26 @@ public int getCount() {
...
+        public void setUsersTotal(String total) {
+            Log.d("page total", total + "");
+            newTitles[USERS_FRAGMENT] = " (" + total + ")";
+            adapter.notifyDataSetChanged();
+        }
     }
 
     public interface SearchQueryListener {
         void search(String query);
+
```
Comment (Dima564) [inline app/src/main/java/com/uptech/example/vidme/ui/search/SearchActivity.java]: 👾  Remove 👾

File: app/src/main/java/com/uptech/example/vidme/model/Page.java [hunk 5]
```diff
@@ -0,0 +1,12 @@
+package com.uptech.example.vidme.model;
+
+/**
+ * Created by bogdan on 19.04.2016.
+ */
```
Comment (Dima564) [inline app/src/main/java/com/uptech/example/vidme/model/Page.java]: Use Lombok

File: app/src/main/java/com/uptech/example/vidme/api/dto/VideoListResponse.java [hunk 6]
```diff
@@ -8,10 +9,16 @@
  * Created by oleh on 04.04.16
  */
 public class VideoListResponse {
-//    private boolean status;
+    //    private boolean status;
```
Comment (Dima564) [inline app/src/main/java/com/uptech/example/vidme/api/dto/VideoListResponse.java]: Use Lombok

File: app/src/main/java/com/uptech/example/vidme/api/dto/UserResponse.java [hunk 7]
```diff
@@ -12,10 +13,15 @@
     private Auth auth;
     private User user;
     private List<User> users;
+    private Page page;
 
     public UserResponse() {
     }
```
Comment (Dima564) [inline app/src/main/java/com/uptech/example/vidme/api/dto/UserResponse.java]: Lombok

File: app/src/main/java/com/uptech/example/vidme/adapters/VideoSearchListAdapter.java [hunk 8]
```diff
@@ -53,6 +53,16 @@ public void onBindViewHolder(ViewHolder holder, int position) {
...
+            @Override
+            public void onClick(View v) {
+                if (!holder.likeClicked) {
+                    holder.likesCount.setText(String.valueOf(Integer.parseInt(holder.likesCount.getText().toString()) + 1));
+                    holder.likeClicked = true;
```
Comment (AndriyBas) [inline app/src/main/java/com/uptech/example/vidme/adapters/VideoSearchListAdapter.java]: @bogdanStepanyuk we should not create listeners in `getView()`

File: app/src/main/java/com/uptech/example/vidme/adapters/VideoSearchListAdapter.java [hunk 9]
```diff
@@ -79,6 +89,7 @@ public Video getItem(int position) {
         protected TextView viewsCount;
         @Bind(R.id.likes_count)
         protected TextView likesCount;
+        private boolean likeClicked;
```
Comment (AndriyBas) [inline app/src/main/java/com/uptech/example/vidme/adapters/VideoSearchListAdapter.java]: @bogdanStepanyuk use model, not holder for this (as discussed)

File: app/src/main/java/com/uptech/example/vidme/ui/search/SearchActivity.java [hunk 10]
```diff
@@ -33,8 +39,12 @@
     TabLayout tab_layout;
     @Bind(R.id.view_pager_search)
     ViewPager viewPager;
+    @Bind(R.id.edit_search)
+    EditText editSearch;
```
Comment (AndriyBas) [inline app/src/main/java/com/uptech/example/vidme/ui/search/SearchActivity.java]: @bogdanStepanyuk please use access modifiers

File: app/src/main/java/com/uptech/example/vidme/ui/search/SearchActivity.java [hunk 11]
```diff
@@ -50,52 +60,87 @@ protected void onCreate(Bundle savedInstanceState) {
...
+            imm.hideSoftInputFromWindow(editSearch.getWindowToken(), 0);
+        }
+    }
 
-        return super.onCreateOptionsMenu(menu);
+    private void openKeyboard() {
+        if (editSearch != null) {
+            InputMethodManager imm = (InputMethodManager) getSystemService(Context.INPUT_METHOD_SERVICE);
+            imm.showSoftInput(editSearch, InputMethodManager.SHOW_IMPLICIT);
+        }
```
Comment (AndriyBas) [inline app/src/main/java/com/uptech/example/vidme/ui/search/SearchActivity.java]: @bogdanStepanyuk please pass `EditText` as argument to methods `hideKeyboard()` and `openKeyboard()` to make them more extensible

File: app/src/main/java/com/uptech/example/vidme/ui/search/SearchActivity.java (hunk 3 above)
Comment (AndriyBas) [inline app/src/main/java/com/uptech/example/vidme/ui/search/SearchActivity.java]: @Dima564 @bogdanStepanyuk better add `.trim()` for string. In general, always `trim` all input from user

**Pull request merged**